### ISSUE TITLE
The ZHA integration page should list its IoT class as "Local Push" instead of showing the IoT class as "Local Polling"

### ISSUE THREAD

**Issue by Hedda** [home-assistant/home-assistant.github.io#26125]
### Feedback

The ZHA integration page should list its IoT class as "Local Push" instead of showing the IoT class as "Local Polling".

Wondering why IoT class classification of [ZHA integration](https://www.home-assistant.io/integrations/zha) only listed as being "Local Polling" when it really is more "Local Push"? 

As I understand it "Local Polling" or also "Local Push" is per device in Zigbee(?), and especially older devices (with old firmware / old Zigbee stacks) used to require pulling, but most newer Zigbee devices push their reports to the Zigbee Coordinator?

Note that this IoT classofication is also a marketing thing, as many users prefer integrations that uses "Local Push" instead of "Local Polling" and may avoid not even try such integrations if it sounds worse than it is. 😉 

The question is if the ZHA integration is today really only "Local Polling" or only "Local Push", or does it in fact support both and it depend on the devices being used and attributes being read? So should maybe say "Local Polling and Local Push"?

![2023-02-05 (2)](https://user-images.githubusercontent.com/6320001/216840297-a87a14ba-e0b9-4bad-82c9-cb714d11544e.png)

I thought Zigbee supported both "Local Push" and "Local Polling" but that it will depend on what each device user uses supports?

https://www.home-assistant.io/blog/2016/02/12/classifying-the-internet-of-things/#classifiers

* **Local Polling** = "**_Offers direct communication with device. Polling the state means that an update might be noticed._**"

* **Local Push** =  "**_Offers direct communication with device. Home Assistant will be notified as soon as a new state is available._**"

https://www.home-assistant.io/blog/2016/02/12/classifying-the-internet-of-things/#polling-the-local-device

### POLLING THE LOCAL DEVICE

These devices will offer an API that is locally accessible. The home automation will have to frequently check if the state has been updated.

Advantages:
* Does not depend on the internet.

Disadvantages:
* To be pollable, a device needs to be always online which requires the device to be connected to a power source.

https://www.home-assistant.io/blog/2016/02/12/classifying-the-internet-of-things/#local-device-pushing-new-state

###  LOCAL DEVICE PUSHING NEW STATE

The best of the best. These devices will send out a notice when they get to a new state. These devices usually use a home automation protocol to pass it’s message to a hub that will do the heavy lifting of managing and notifying subscribers

Advantages:
* Near instant delivery of new states.
* Able to get a long battery life by going into deep sleep between state updates.

Disadvantages:
* If it does not also support polling, home automation will not be made aware of the state after booting up until it changes.
* If using deep sleep and Wi-Fi, it will suffer a delay when waking up because connecting to WiFi and receiving an IP takes time.

### URL

https://www.home-assistant.io/integrations/zha

### Additional information

To compare; Z-Wave JS integration is classified as "Local Push", and Z-Wave uses very similar architecture/technology to Zigbee:

* https://www.home-assistant.io/integrations/zwave_js

![2023-02-05 (3)](https://user-images.githubusercontent.com/6320001/216840780-c3ba7e17-f38f-4889-997f-498f22999706.png)

So if can only pick one or the other and Z-Wave which use similar protocol says "Local Push", would it be better to say same?

**Comment by Hedda** [comment]
Still open unanswered question

**Comment by Hedda** [comment]
Not stale. Still valid question I think.

**Comment by Hedda** [comment]
Still wondering about this. Interestingly, IoT class for the the new [Matter integration](https://www.home-assistant.io/integrations/matter/) is listed as "Local Push" while the [Thread integration](https://www.home-assistant.io/integrations/thread) and the [Open Thread Border Router integration](https://www.home-assistant.io/integrations/otbr/) are listed as "Local Polling", though believe all probably do both?

**Comment by Hedda** [comment]
Still very relavant since users are confused by this and think the ZHA integration will be slow because it says it using pulling.

Again, does the ZHA integration not use both? 

You use push when you change state, like a switch to ON or OFF, and then use pull when you check something like power monitoring.

It is confusing that the ZHA integration say that it uses pulling if it also uses pushing too depending on the command.

People associate pulling with slow and delays.

**Comment by codyc1515** [comment]
I agree with you actually. Should be Local Push, based on my knowledge of Zigbee and how it operates.

**Comment by Hedda** [comment]
Still not answered. Again, I do believe different Zigbee devices use "Local Polling" and others use "Local Push" or a combination.

https://www.waveshare.com/w/upload/4/4b/Zigbee-Wireless-Networking.pdf

_ZCL allows for both “ push ” and “ pull ” methods (report attribute and read attribute methods), for determining the state of any given cluster’s attributes. Take a look at Figure 6.4 ._

_In the “ push” method, the devices themselves send reports when the attribute changes. The report can be configured to be time-based, change-in-value-based, or both. For example, an on/off light could be configured to report every time the light changes state (on or off). A temperature sensor could be configured to report once every minute, or if the temperature raises or lowers by five degrees, whichever comes first. A settable minimum reporting interval ensures that if the temperature flutters between two values, a report isn’t sent too often._

_In the “ pull” method, the device that needs the information asks the other devices for their current values. For example, when someone picks up the TV remote control, the current outside temperature could be read from the external temperature sensor, and reported on the remote’s display or the television screen. A Home Automation management system could ask each node for the manufacturer information and display a systemwide view each time a PC application starts up._

![image](https://github.com/home-assistant/home-assistant.io/assets/6320001/f3b1c83e-d00a-4abc-94f1-4a664caa9043)

**Comment by codyc1515** [comment]
Still current.

**Comment by Hedda** [comment]
Bump as it would be good for highlighting the **ZHA integration** for new users if the page said "Local Polling" instead of "Local Push".

Again, please compare this to the page for the **Z-Wave integration** and **Matter integration** that their IoT class is [Local Push.](https://www.home-assistant.io/blog/2016/02/12/classifying-the-internet-of-things/#classifiers), see:

* https://www.home-assistant.io/integrations/zha

* https://www.home-assistant.io/integrations/zwave_js/

* https://www.home-assistant.io/integrations/matter/

<img width="665" height="664" alt="Image" src="https://github.com/user-attachments/assets/a86fe303-a16b-413e-baa2-de25b358a9d4" />

<img width="642" height="624" alt="Image" src="https://github.com/user-attachments/assets/872011aa-b3f3-4cd2-a090-ec42f15f4ec9" />

<img width="673" height="636" alt="Image" src="https://github.com/user-attachments/assets/51408d52-9022-4fc4-97ec-958eab8ad3f6" />
